docs/software_requirements/queues.sdoc

```diff
@@ -0,0 +1,216 @@
+[DOCUMENT]
+TITLE: Tracing
+REQ_PREFIX: ZEP-SRS-20-
+
+[GRAMMAR]
+IMPORT_FROM_FILE: software_requirements.sgra
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-1
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue definition at compile time
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to define and initialize a queue at compile time.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-2
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue definition at run time.
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to define and initialize a queue at run time.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-3
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue append item to the end
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to append a data item to the end of a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-4
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue append item to the start
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to append a data item in the first place of a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-5
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue remove item
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to remove a data item from a queue.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-21
+
+[REQUIREMENT]
+UID: ZEP-SRS-20-6
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Queues
+TITLE: Queue get item
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to get a data item from a queue.
```

Comment:
To me the "to get a data item" is not fully unambiguous - does this mean "to read", "to obtain and remove", "to read and then remove", "to remove" ? I have no better expression to suggest, though. Would be ok for me, if the majority here says the meaning of "to get" is clear in this context.

Reply:
corrected it to "get and remove"